java/code/src/com/suse/manager/reactor/PGEventStream.java

```diff
@@ -123,17 +123,13 @@ public void run() {
                     try (Statement s = connection.createStatement()) {
                         s.execute("SELECT 'salt-event-connection-watchdog';");
 
-                        // if we have any rows in suseSaltEvent that do not yet have a process task queued or executing
+                        // if we have any rows in suseSaltEvent that do not yet have a process task active
                         // then schedule tasks for them
                         // this can only happen in case we lost notifications somehow
                         List<Long> allJobs = SaltEventFactory.countSaltEvents(THREAD_POOL_SIZE + 1);
 
-                        List<Long> queuedJobs = executorServices.stream()
-                                .map(executor -> executor.getTaskCount() - executor.getCompletedTaskCount())
-                                .collect(Collectors.toList());
-
                         List<Long> missingJobs = IntStream.range(0, allJobs.size())
-                            .mapToObj(i -> allJobs.get(i) - queuedJobs.get(i))
+                            .mapToObj(i -> executorServices.get(i).getActiveCount() > 0 ? 0 : allJobs.get(i))
```

Comment:
@nadvornik could you explain the logic behind this if statement? Why if active count is bigger then 0 the missing jobs are 0? The all jobs should not be taken in consideration also?

Reply:
Active count is bigger then 0 means that the thread is running and processing events.
So if the thread is running, then everything is OK and no action is needed.
When the thread stopped and there are queued events - it means that something went wrong and the best we can do is to schedule all queued events.